frontend/src/components/managePermissions/table/DelegatedAdminTable.vue

```diff
@@ -0,0 +1,126 @@
+<script setup lang="ts">
+import { ref, type PropType } from 'vue';
+import { FilterMatchMode } from 'primevue/api';
+import Column from 'primevue/column';
+import DataTable from 'primevue/datatable';
+
+import { routeItems } from '@/router/routeItem';
+import DataTableHeader from '@/components/managePermissions/table/DataTableHeader.vue';
+import type { FamAccessControlPrivilegeGetResponse } from 'fam-admin-mgmt-api/model';
+
+const props = defineProps({
+    loading: {
+        type: Boolean,
+        default: false,
+    },
+    delegatedAdmins: {
+        type: [Array] as PropType<FamAccessControlPrivilegeGetResponse[] | undefined>,
+        required: true,
+    },
+});
+
+const delegatedAdminFilters = ref({
+    global: { value: '', matchMode: FilterMatchMode.CONTAINS },
+    'user.user_name': { value: null, matchMode: FilterMatchMode.STARTS_WITH },
+    'user.user_type.description': {
+        value: null,
+        matchMode: FilterMatchMode.CONTAINS,
+    },
+    'role.role_name': {
+        value: null,
+        matchMode: FilterMatchMode.CONTAINS,
+    },
+    'role.parent_role.role_name': {
+        value: null,
+        matchMode: FilterMatchMode.CONTAINS,
+    },
+});
+
+const delegatedAdminSearchChange = (newvalue: string) => {
+    delegatedAdminFilters.value.global.value = newvalue;
+};
+</script>
+
+<template>
+    <!-- Hidden until functionality is available
+    <ConfirmDialog group="deleteAdmin">
+        <template #message>
+            <ConfirmDialogtext
+                :userName=""
+                :role=""
+            />
+        </template>
+    </ConfirmDialog> -->
+    <div class="data-table-container">
+        <div class="custom-data-table">
+            <DataTableHeader
+                btnLabel="Create delegated admin"
+                :btnRoute="routeItems.grantAppAdmin.path"
+                :filter="delegatedAdminFilters['global'].value"
+                @change="delegatedAdminSearchChange"
+            />
+            <DataTable
+                v-model:filters="delegatedAdminFilters"
+                :value="props.delegatedAdmins"
+                paginator
+                :rows="50"
+                :rowsPerPageOptions="[5, 10, 15, 20, 50, 100]"
+                filterDisplay="menu"
+                :loading="props.loading"
+                :globalFilterFields="[
+                    'user.user_name',
+                    'user.user_type.description',
+                    'role.role_name.role_name',
+                    'role.parent_role.role_name',
+                ]"
+                paginatorTemplate="RowsPerPageDropdown CurrentPageReport PrevPageLink NextPageLink"
```

Comment:
we use this paginatorTemplate and the currentPageReportTemplate in all three tables, let's put it in the constant file

Reply:
sure, I will add it to a constant